programs/loader-v4/src/lib.rs

```diff
@@ -104,15 +97,31 @@ pub fn load_program_from_account(
         aligned_memory_mapping: true,
         // Warning, do not use `Config::default()` so that configuration here is explicit.
     };
-    let loader = BuiltinProgram::new_loader(config);
+    BuiltinProgram::new_loader(config)
+}
+
+pub fn load_program_from_account(
+    _feature_set: &FeatureSet,
+    compute_budget: &ComputeBudget,
+    log_collector: Option<Rc<RefCell<LogCollector>>>,
+    program: &BorrowedAccount,
+    debugging_features: bool,
+) -> Result<(Arc<LoadedProgram>, LoadProgramMetrics), InstructionError> {
+    let mut load_program_metrics = LoadProgramMetrics {
+        program_id: program.get_key().to_string(),
+        ..LoadProgramMetrics::default()
+    };
     let state = get_state(program.get_data())?;
     let programdata = program
         .get_data()
         .get(LoaderV4State::program_data_offset()..)
         .ok_or(InstructionError::AccountDataTooSmall)?;
     let loaded_program = LoadedProgram::new(
         &loader_v4::id(),
-        Arc::new(loader),
+        Arc::new(create_program_runtime_v2_environment(
```

Comment:
This entire function needs to be moved inside `Bank::load_program()` and then use `loaded_programs_cache.program_runtime_environment_v2`. But that can be done in a different PR.

Reply:
Yes, that's in my other PR, where it's integrating it to `bank::load_program`.